src/algebra/group_ring_action.lean

```diff
@@ -65,6 +65,19 @@ def mul_semiring_action.to_ring_equiv [mul_semiring_action G R] (x : G) : R ≃+
 { .. distrib_mul_action.to_add_equiv R x,
   .. mul_semiring_action.to_ring_hom G R x }
 
+section
+variables {M N}
+
+/-- Compose a `mul_semiring_action` with a `monoid_hom`, with action `f r' • m`.
+See note [reducible non-instances]. -/
+@[reducible] def mul_semiring_action.comp_hom (f : N →* M)
+[mul_semiring_action M R] : mul_semiring_action N R :=
```

Comment:
One last change: this line can now be merged with the previous one

Reply:
Done